cdap-security/src/test/java/co/cask/cdap/security/authorization/InMemoryAuthorizer.java

```diff
@@ -168,24 +170,32 @@ public void removeRoleFromPrincipal(Role role, Principal principal) throws RoleN
         privileges.addAll(getPrivileges(role));
```

Comment:
Role is also a Principal with PrincipalType.ROLE. So someone could call grant on a Role, and that would update the privileges map The roleToPrincipals I think was only to maintain the membership between roles and users/groups.